File: frontend/src/types/project.ts
export interface Project {
  id: number
  master_theme_id: number
  name: string
  status: string
  clip_count: number
  music_count: number
  target_duration: number
  image_provider: string
  video_provider: string
  music_provider: string
  owner_id: number | null
  category_id: number | null
  created_at: string
  updated_at: string
}

export interface ProjectInput {
  master_theme_id: number
  name: string
  clip_count?: number
  music_count?: number
  target_duration: number
  category_id?: number | null
}

export interface ActivityIdeas {
  activities: string[]
}

export interface ProjectReviewSummary {
  project_id: number
  project_name: string
  category_name: string | null
  completed_video_count: number
}

File: backend/app/schemas/project.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class ProjectBase(BaseModel):
    master_theme_id: int
    name: str
    clip_count: int = Field(default=0, ge=0)
    music_count: int = Field(default=0, ge=0)
    target_duration: int = Field(gt=0)
    image_provider: str = "openai"
    video_provider: str = "kling"
    music_provider: str = "manual"
    category_id: int | None = None


class ProjectCreate(ProjectBase):
    pass


class ProjectRead(ProjectBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    status: str
    owner_id: int | None
    created_at: datetime
    updated_at: datetime


class ProjectReviewSummary(BaseModel):
    project_id: int
    project_name: str
    category_name: str | None
    completed_video_count: int
